pkg/azuredx/client/httpclient.go

```diff
@@ -81,6 +81,11 @@ func getAuthOpts(azureSettings *azsettings.AzureSettings, dsSettings *models.Dat
 		if err != nil {
 			return nil, err
 		}
+
+		if dsSettings.AllowUserTrustedEndpoints && len(dsSettings.UserTrustedEndpoints) > 0 {
+			endpoints = append(endpoints, dsSettings.UserTrustedEndpoints...)
```

Comment:
The method getAdxEndpoints has a limitation where it doesn't work for custom clouds provided through the [clouds_config setting](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#clouds_config) because it throws an error if the cloud name is not known.  I think that means that this feature won't work for custom clouds.  I think we will need to re-work getAdxEndpoints so that it does not throw for an unknown clouds.

Reply:
Looks good - thanks!

Reply:
@aangelisc safe to assume this will only work on 12.x bits?  Have you tested these changes on 11.x to make sure they are back compatible?